torch/csrc/jit/codegen/cuda/glfdc/eval.cpp

```diff
@@ -0,0 +1,336 @@
+#include <torch/csrc/jit/codegen/cuda/glfdc/eval.h>
+
+#include <torch/csrc/jit/codegen/cuda/glfdc/cfold.h>
+
+using namespace torch::jit::fuser::cuda::glfdc;
+
+namespace {
+
+using skiptree_t = std::vector<std::pair<SExprRef, std::size_t>>;
```

Comment:
Please add a comment here. What does each pair represent?

Reply:
I think this type is basically a tree structure flattened out on a vector. In each vector element `skip_tree[i]`, `skip_tree[i].first` is a reference to the actual tree/IR node, and `skip_tree[i].second` is the incremental index on the vector. For example, the indexes of all child nodes of `skip_tree[i].first` on `skip_tree` will be the nodes between `i` and `i+skip_tree[i].second`.

@prak-nv please correct if not accurate and please do include comments explaining this data structure.

Reply:
Yes, skip_tree has same structure as list of operations. In other words is skip_tree[i].second is number of childs in subtree where skip_tree[i].first is root node.